### DISCUSSION TITLE
The future of Scene Manager

### DISCUSSION THREAD

**Discussion by jabsatz** [glass-brick/Scene-Manager#35] [Announcements]
## Hello!

I have been quite absent for a while and thought that writing a blog post here would be the best place to address the state of this plugin going forward.

As some may have gathered from the slow updates to this plugin during the past year, I haven't quite been active in development of this plugin. This is mostly because of bad management on my side, going in I made the whole thing in a couple of days while I was very active in several different game jam events, and found that a plugin like this was a great time-saver for these kinds of projects.

I am not a full-time godot developer and this is not a very big plugin (neither in amount of users nor in scope), so I tend to forget to do work on it if I'm not actively involved in something, this is the main reason why the plugin has stagnated a bit.

Even with that being the case, I still think this plugin is worthwhile for small and medium projects alike, and I would love for it to remain in a usable state.

I have released a new version (1.1.0) which should be shortly available in the godot asset library as well. It includes a couple of PRs that other contributors fixed (HUGE thanks to @JEuler, @zen14774 and @myin142!!), hopefully with these issues solved the plugin is stable in the features it offers.

As for future features, the main one I've read was mentioned here https://github.com/glass-brick/Scene-Manager/issues/33. Background loading is a feature that I actually envisioned adding when I was putting the plugin together, however since I had little experience on godot at the time, and frankly no need to use it myself, I kind of forgot about it. That said, it's something I'd love to have added since it's the perfect place to implement it and I would like to say "I'll get to it", but seeing my output during the past year, it seems unlikely that I will.

There is also a bug mentioning a memory leak here https://github.com/glass-brick/Scene-Manager/issues/34, but the setup for it does not seem like something I would be able to reproduce and fix quickly either.

This is a bit of a long-winded way of saying that I will probably not be personally fixing further issues that do not hinder the most common usecases of the plugin. But, I am also not planning to leave this project completely to disappear. In order to ensure that, I'll be committing to the following:

**I will continue monitoring this project periodically, at least answering to issues and such, and if any further PRs are opened, I will personally review them, take care of merging and releasing new versions both here and in the godot asset library.**

With that in mind, going forward I will be relying more on community fixes and features. This is not exactly news, as exemplified by the fact most of the past few commits have been made by other contributors, but I want to come clean about it.

A reminder, again, that **PRs are open for anybody**. The only thing I ask is that you try to create an issue first, specially if it's a new feature, since I wouldn't want to increase the scope of this plugin too much.

Well, that's basically all I wanted to say. I will take a look at the open issues right now and tag them/close them accordingly to the plans described here to keep the repository in a clean state.

**Comment by zen14774** [comment]
Thanks to you for the plugin. To me it goes into the "does one thing and does it well, download and forget" category. As long as fixes get merged and it keeps up with Godot releases it's all I would ask for.
